### DISCUSSION TITLE
Possible to fade lights over a period of hours?

### DISCUSSION THREAD

**Discussion by probonopd** [arendst/Tasmota#12989] [Q&A] [answered]
Is it possible to slowly, smoothly fade/change lights over a period of hours?

My use case is to implement [circadian lighting](https://www.thelightingpractice.com/what-is-circadian-lighting/).

![image](https://user-images.githubusercontent.com/2480569/131021347-57152062-26b5-49d7-81b3-8f18e18c3285.png)

I would like to say, e.g., at 6 am the light is at 20% brightness and 2700 Kelvin, at 12pm it is 100% brightness and 6500 Kelvin - and in between, it fades smoothly over the whole time so that the change is so slow that it cannot be noticed.

Is this currently possible?

**Comment by barbudor** [comment, marked as answer]
You can use [Timers](https://tasmota.github.io/docs/Timers/) and [Rules](https://tasmota.github.io/docs/Rules/)
With timers, you can have up to 16 events programmed, each triggering a rule which set the dimmer to a given value
Timers and Rules are available as default features in Tasmota.bin

With [Scripting-Language](https://tasmota.github.io/docs/Scripting-Language/) it may be possible to even implement a formula instead of fixed values. Scripting-Language is not a default feature and requires you to [compile](https://tasmota.github.io/docs/Compile-your-build/) your own binary.

**Reply by sfromis** [reply]
You could also setup a self-restarting sequence of  RuleTimer commands, nudging the light a bit each cycle.

**Reply by probonopd** [reply]
Thanks, I figured that timers can be used to set the light to certain values at certain points in time. But what I am missing is how to make it transition smoothly over the whole time. In my example, I don't want the brightness to jump from 20% to 100% all of a sudden. But instead 20.000%, 20.001%, 20.002%...veeeeery slow changes over hours of time. Ideally I would just specify how the light should look at certain points in time (timers) and everything in between would be automatically interpolated.

**Reply by techman83** [reply]
Pretty much as @sfromis said, all the tools are there. You could have a timer that adjusts a variable, then have the variable trigger the change. I'm not sure how much precision you can store in a var though, that'd be an experiment left up to the reader :) 

The doco has a lot of good examples you can comb through.
https://tasmota.github.io/docs/Rules/#setting-variables
https://tasmota.github.io/docs/Rules/#automatically-vary-the-color-temperature-of-a-cct-light

**Reply by probonopd** [reply]
Thank you, this is exactly what I was looking for :+1:
